client-js/client/rest_helpers.ts

```diff
@@ -0,0 +1,88 @@
+import * as RTVIErrors from "../rtvi/errors";
+import { logger } from "./logger";
+import { TransportConnectionParams } from "./transport";
+
+export interface ConnectionEndpoint {
+  endpoint: string;
+  headers?: Headers;
+  requestData?: object;
```

Comment:
`object` is a fairly open type, as it allows anything where `typeof` is `object`, so even things like `new Date()`, `new RegExp()`, `new WeakMap()` etc.
Do we want to narrow this a bit down? Or is this intentionally open, because the requestData can be literally anything?

Reply:
`object` doesn't allow primitives, like numbers, strings or booleans, which are generally JSON-serializable.
I think it's a good idea to inform about the intent of `requestData` ("should be JSON-serializable") and provide a matching type:

```ts
type Serializable =
  | string
  | number
  | boolean
  | null
  | Serializable[]
  | { [key: number | string]: Serializable };
```

Unless we _require_ `requestData` to be an object, then we could set it to:

```ts
type SerializableRequestData = { [key: number | string]: Serializable };
```

Reply:
thanks!